proposals/0475-tuple-syntax.rst

```diff
@@ -265,41 +277,41 @@ Proposed Change Specification
       synonym for ``GHC.Exts.Sum<n># ty1 ty2 ... ty<n-1> ty<n>``.
 
    #. An occurrence of ``()`` in type-syntax, where the type is expected to be of kind ``Constraint``,
-      is a synonym for ``GHC.Tuple.CUnit``.
+      is a synonym for ``GHC.Classes.CUnit``.
 
    #. An occurrence of ``(ty1, ty2, ..., ty<n-1>, ty<n>)`` (for *n* ≧ 2) in type-syntax, where the type is
-      expected to be of kind ``Constraint``, is a synonym for ``GHC.Tuple.CTuple<n> ty1 ty2 ... ty<n-1> ty<n>``.
+      expected to be of kind ``Constraint``, is a synonym for ``Data.Tuple.Experimental.CTuple<n> ty1 ty2 ... ty<n-1> ty<n>``.
 
    #. An occurrence of ``(ty1, ty2, ..., ty<n-1>, ty<n>)`` (for *n* ≧ 2) in type-syntax, where the first
       ``ty<i>`` inferred to have kind ``Type`` or ``Constraint`` has kind ``Constraint``, is a synonym
-      for ``GHC.Tuple.CTuple<n> ty1 ty2 ... ty<n-1> ty<n>``.
+      for ``GHC.Classes.CTuple<n> ty1 ty2 ... ty<n-1> ty<n>``.
 
    #. An unapplied occurrence of ``GHC.List.List`` is pretty-printed as ``[]``.
 
    #. An occurrence of ``GHC.List.List ty`` is pretty-printed as ``[ty]``.
 
-   #. An occurrence of ``GHC.Tuple.Unit`` is pretty-printed as ``()``.
+   #. An occurrence of ``GHC.Tuple.Prim.Unit`` is pretty-printed as ``()``.
 
-   #. An occurrence of ``GHC.Tuple.Tuple<n> ty1 ty2 ... ty<n>`` is pretty-printed as ``(ty1, ty2, ..., ty<n>)``.
+   #. An occurrence of ``GHC.Tuple.Prim.Tuple<n> ty1 ty2 ... ty<n>`` is pretty-printed as ``(ty1, ty2, ..., ty<n>)``.
 
-   #. An occurrence of ``GHC.Tuple.Tuple<n>``, but not applied to a full *n* arguments, is pretty-printed as ``(,,...,,)``,
+   #. An occurrence of ``GHC.Tuple.Prim.Tuple<n>``, but not applied to a full *n* arguments, is pretty-printed as ``(,,...,,)``,
       where there are *n-1* commas.
 
-   #. An occurrence of ``GHC.Exts.Unit#`` is pretty-printed as ``(# #)``.
+   #. An occurrence of ``GHC.Types.Unit#`` is pretty-printed as ``(# #)``.
```

Comment:
those should all have been changed to `GHC.Types`, I overlooked a few occurrences.
`GHC.Prim` is a virtual module, so my assumption is that Richard hadn't determined the precise module yet at the point of writing it.

The straightforward change would have been to use `Data.Tuple.Experimental` instead of `GHC.Exts` – I used `GHC.Types` because I now know what we landed on eventually…do you think that's too much retconning? We could also change the other sentence to say "would likely be exported from `GHC.Types` originally".

Reply:
ah no, there's no equivalent of `GHC.Exts` with the new schema…